### PR TITLE
Remove support for non-dedicated cc's on hosted

### PR DESCRIPTION
**Pull request by bratseth**

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java [hunk 1]
```diff
@@ -238,6 +239,14 @@ private HostSpec retire(HostSpec host) {
...
     }
 
+    private void dumpFreeNodes(String description) {
+        System.out.println("  Free nodes " + description + ":");
+        for (var entry : freeNodes.entrySet()) {
+            for (var node : entry.getValue())
+                System.out.println("    " + node.hostname());
+        }
+    }
+
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java:249]: Leftover?

File: config-model/src/main/java/com/yahoo/vespa/model/content/cluster/ContentCluster.java [hunk 2]
```diff
@@ -293,26 +293,8 @@ private void addClusterControllers(ConfigModelContext context,
             if (contentCluster.getPersistence() == null) return;
 
             ClusterControllerContainerCluster clusterControllers;
-
-            ContentCluster overlappingCluster = findOverlappingCluster(context.getParentProducer().getRoot(), contentCluster);
-
-            if (overlappingCluster != null && overlappingCluster.getClusterControllers() != null) {
-                // Borrow the cluster controllers of the other cluster in this case.
-                // This condition only occurs on non-hosted systems with a shared config server,
-                // a combination which only exists in system tests.
```
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/vespa/model/content/cluster/ContentCluster.java:302]: You dare challenge this comment?

Comment (jonmv) [reply config-model/src/main/java/com/yahoo/vespa/model/content/cluster/ContentCluster.java:302]: Ah, never mind, I'm just bad at reading.

**Review by jonmv — APPROVED** [review]
Great!

File: config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java (hunk 1 above)
Comment (jonmv) [inline config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java:249]:
```suggestion
```

Comment (bratseth) [reply config-model/src/main/java/com/yahoo/config/model/provision/InMemoryProvisioner.java]: Thought I'm going to need it later, but ok :-)

**Pull request merged**